Interpred tools test

Cidana-Developers opened the pull request.

andrey-norkin commented on test/ref/EbComputeMean_C.c:
```diff
@@ -0,0 +1,126 @@
+    uint32_t inputAreaWidth,   // input parameter, input area width
+    uint32_t inputAreaHeight)  // input parameter, input area height
+{
+    uint32_t horizontalIndex;
```
Variables in camelCase. Should be in snake_case.

Cidana-Developers replied:
Done.

andrey-norkin commented on test/ref/EbComputeMean_C.c:
```diff
@@ -0,0 +1,126 @@
+ *******************************************/
+uint64_t ComputeMean(
+    uint8_t* inputSamples,     // input parameter, input samples Ptr
+    uint32_t inputStride,      // input parameter, input stride
```
variables in camel case.

Cidana-Developers replied:
Done.

andrey-norkin commented:
There are some warnings in Unit Test build

Cidana-Developers commented:
> There are some warnings in Unit Test build

Done, its clean now.

andrey-norkin commented:
The Unit Test build on Linux passes without warnings now but the osx build still has some.

Cidana-Developers commented:
> The Unit Test build on Linux passes without warnings now but the osx build still has some.

Rebased and removed the ld warnings of osx build and other warnings introduced recently.

Cidana-Developers commented:
Regarding the failed cases (HmeTest and SadLoopSparseTest), they can be fixed by PR #444 and #458.

mafonsoN commented:
@Cidana-Developers can you rebase and fix the conflicts?

Cidana-Developers commented:
> @Cidana-Developers can you rebase and fix the conflicts?

Sure, it's done.

mafonsoN commented:
@hassount any concerns about merging these tests?

hassount commented on CMakeLists.txt:
```diff
@@ -69,7 +69,6 @@ set(flags_to_test
     -Wformat
     -Wformat-security
     -fstack-protector-strong
-    -fPIE
```
why did we remove this cmake flag ?

Cidana-Developers replied:
It's removed when we tried to eliminate the link warning on building shared libraries on osx:
`ld: warning: -pie being ignored. It is only used when linking a main executable`.
Will rebase it and add it back.

hassount replied:
thank you

hassount commented:
> @hassount any concerns about merging these tests?

It's looks fine except for the compiler flag removed. it is needed as part of the requirements of our security team.
